src/taipy/gui/gui.py

```diff
@@ -445,18 +452,24 @@ def __call_on_change(self, var_name: str, value: t.Any, on_change: t.Optional[st
             var_name = var_name_decode
         else:
             if var_name not in self.__var_dir._var_head:
-                warnings.warn(f"Can't find matching variable for {var_name} on {current_context} context")
-                return
+                raise NameError(f"Can't find matching variable for {var_name} on {current_context} context")
             _found = False
             for k, v in self.__var_dir._var_head[var_name]:
                 if v == current_context:
                     var_name = k
                     _found = True
                     break
             if not _found:
-                warnings.warn(f"Can't find matching variable for {var_name} on {current_context} context")
-                return
-        var_name = f"{var_name}.{suffix_var_name}" if suffix_var_name else var_name
+                raise NameError(f"Can't find matching variable for {var_name} on {current_context} context")
+        return f"{var_name}.{suffix_var_name}" if suffix_var_name else var_name, current_context
+
+
+    def __call_on_change(self, var_name: str, value: t.Any, on_change: t.Optional[str] = None):
+        try:
+            var_name, current_context = self.__get_real_var_name(var_name)
+        except Exception as e:
+            warnings.warn(f"{e}")
```

Comment:
Is a warning enough at this point? May be an actual error?

Reply:
I didn't touch this part (only refactor)